modules/imgproc/src/contours_common.hpp

```diff
@@ -159,23 +171,78 @@ class TreeIterator
     }
 
 private:
-    std::stack<int> levels;
     Tree<T>& tree;
+    std::stack<int> levels;
 };
 
 //==============================================================================
 
+template <typename T, size_t BLOCK_SIZE_ELEM, size_t STATIC_CAPACITY_BYTES>
+class ContourDataStorage
+{
+public:
+    typedef T data_storage_t;
+    typedef BlockStorage<data_storage_t, BLOCK_SIZE_ELEM, STATIC_CAPACITY_BYTES> storage_t;
+public:
+    ContourDataStorage(void) = delete;
+    ContourDataStorage(storage_t* _storage):storage(_storage) {}
+    ContourDataStorage(const ContourDataStorage&) = delete;
+    ContourDataStorage(ContourDataStorage&&) noexcept = default;
+    ~ContourDataStorage() = default;
+    ContourDataStorage& operator=(const ContourDataStorage&) = delete;
+    ContourDataStorage& operator=(ContourDataStorage&&) noexcept = default;
+public:
+    typename storage_t::RangeIterator getRangeIterator(void) const {return storage->getRangeIterator(first, last);}
+public:
+    bool empty(void) const {return first == last;}
+    size_t size(void) const {return last - first;}
+public:
+    void clear(void) {first = last;}
+    bool resize(size_t newSize) {
+        bool ok = (newSize <= size());
+        if (ok)
+            last = first+newSize;
+        return ok;
+    }
+    void push_back(const data_storage_t& value) {
```

Comment:
It would be better to keep formatting consistent in existing files. This file uses Allman brace/indent style:
```
void push_back(...)
{
    if (...)
    {
        ...
    }
    ...
}
```